test/functional/launch/shared_mem/tests/test-launch-StaticMem.hpp

```diff
@@ -0,0 +1,116 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_LAUNCH_STATIC_MEM_HPP__
+#define __TEST_LAUNCH_STATIC_MEM_HPP__
+
+#include <numeric>
+
+template <typename INDEX_TYPE, typename WORKING_RES, typename LAUNCH_POLICY, typename TEAM_POLICY, typename THREAD_POLICY,
+int THREAD_RANGE>
+void LaunchStaticMemTestImpl(INDEX_TYPE block_range)
+{
+
+  INDEX_TYPE thread_range(THREAD_RANGE);
+
+  RAJA::TypedRangeSegment<INDEX_TYPE> outer_range(RAJA::stripIndexType(INDEX_TYPE(0)), RAJA::stripIndexType(block_range));
+  RAJA::TypedRangeSegment<INDEX_TYPE> inner_range(RAJA::stripIndexType(INDEX_TYPE(0)), RAJA::stripIndexType(thread_range));
+
+  camp::resources::Resource working_res{WORKING_RES::get_default()};
+  INDEX_TYPE* working_array;
+  INDEX_TYPE* check_array;
+  INDEX_TYPE* test_array;
+
+  size_t data_len = RAJA::stripIndexType(block_range)*RAJA::stripIndexType(thread_range);
+
+  allocateForallTestData<INDEX_TYPE>(data_len,
+                                     working_res,
+                                     &working_array,
+                                     &check_array,
+                                     &test_array);
+
+
+  for(int b=0; b<RAJA::stripIndexType(block_range); ++b) {
+    for(int c=0; c<RAJA::stripIndexType(thread_range); ++c) {
+      int idx = c + RAJA::stripIndexType(thread_range)*b;
+      test_array[idx] = INDEX_TYPE(idx);
+    }
+  }
+
+  RAJA::launch<LAUNCH_POLICY>
+    (RAJA::LaunchParams(RAJA::Teams(RAJA::stripIndexType(block_range)),
+                        RAJA::Threads(RAJA::stripIndexType(thread_range))),
+     [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
+
+      RAJA::loop<TEAM_POLICY>(ctx, outer_range, [&](INDEX_TYPE bid) {
+
+          //Since we are using custom index type we have to first use a 
+          //type that the device compiler can intialize, we can then use a
+          //pointer to recast the shared memory to our desired type.
+          //This enables us to work around the following warning: 
+          // warning #3019-D: dynamic initialization is not supported for
+          //a function-scope static __shared__ variable within a __device__/__global__ function
+          RAJA_TEAM_SHARED char char_Tile[THREAD_RANGE*sizeof(INDEX_TYPE)];
+          INDEX_TYPE *Tile = &((INDEX_TYPE *)char_Tile)[0];
```

Comment:
If you've cast it it is the pointer type you need, no need to then get a pointer to the first element.
```suggestion
          INDEX_TYPE *Tile = (INDEX_TYPE *)char_Tile;
```